### ISSUE TITLE
Problem: blocking abci server implementation

### ISSUE THREAD

**Issue by tomtau** [crypto-com/chain#804]
Migrate @devashishdxt 's abci-rs to rust-abci? Could it preserve the block commit semantics?

**Comment by devashishdxt** [comment]
Added `tokio` support in https://github.com/devashishdxt/abci-rs.

**Comment by tomtau** [comment]
@devashishdxt does it handle that one needed "blocking" sync?

> there's still going to need to be synchronization around Commit - ie. while Commit is running on the Consensus conn, Mempool and Query conns should be blocked. We can start by letting ABCI devs handle this, but it's something that might belong in the server itself.

If so, you try to port it to rust-abci -- just put it under some "next" or "v2" module and open a PR. It can then be released in a non-breaking way in a new version of rust-abci (the existing server etc. can be marked as deprecated, but it'd be kept for the moment)

**Comment by devashishdxt** [comment]
> does it handle that one needed "blocking" sync?

No. It does not block other connections during `commit` (in its current state). I'll try adding a global lock during `commit`.
